webapp/views.py

```diff
@@ -716,3 +734,62 @@ def sitemap_index():
 
     response.headers["Content-Type"] = "application/xml"
     return response
+
+
+def marketo_describe():
+    return flask.jsonify(marketo_api.describe().json())
+
+
+def marketo_submit():
+    form_fields = {}
+    for key, value in flask.request.form.items():
+        if value:
+            form_fields[key] = value
+
+    # Check honeypot values are not set
+    honeypots = {}
+    honeypots["name"] = flask.request.form.get("name")
+    honeypots["website"] = flask.request.form.get("website")
+    if honeypots["name"] != None and honeypots["website"] != None:
```

Comment:
i think this can be simplified to `if form_fields.get("name") or form_fields.get("website") `, because both `None` and `""` evaluate to False.

Reply:
You might actually be able to `pop` right here too as if you add `None` as a default.

Reply:
I know this looks strange but it's because there is logically a difference between `None` and empty string here. 

1. The first check, checks if we are working with a form that contains honeypots or the legacy ones using recaptcha. 
2. The second that checks for empty string is actually testing if the honeypots have been triggered 

I am tempted to break these checks out into easy to read named simple functions. But that splits the code around more.

Reply:
Since this wasn't clear to João, maybe you could add a comment explaining?

Reply:
Good idea, done.